src/inmanta/data/__init__.py

```diff
@@ -5213,13 +5342,48 @@ async def version_exists(cls, environment: uuid.UUID, version: int) -> bool:
         return True
 
     @classmethod
-    async def get_version(cls, environment: uuid.UUID, version: int) -> Optional["ConfigurationModel"]:
+    async def get_version(
+        cls,
+        environment: uuid.UUID,
+        version: int,
+        *,
+        connection: Optional[asyncpg.connection.Connection] = None,
+    ) -> Optional["ConfigurationModel"]:
         """
         Get a specific version
         """
-        result = await cls.get_one(environment=environment, version=version)
+        result = await cls.get_one(environment=environment, version=version, connection=connection)
         return result
 
+    # TODO: delete?
+    @classmethod
+    async def get_full_version_base(
```

Comment:
This too was used by my initial implementation but is no longer in use. Unless anyone thinks this is something we think could be immediately useful (e.g. for the web console) I'll remove it.